tests/testthat/test-regression.R

```diff
@@ -277,8 +277,17 @@ test_that("Error if too many dummy predictors",
                            "There are fewer observations.")
           })
 
+test_that("Outcome labels", {
+    bank$Overall <- as.factor(bank$Overall)
+    attr(bank$Overall, "label") <- "lbl"
+    result <- suppressWarnings(Regression(Overall ~ Fees + Interest + Phone + Branch + Online + ATM,
+                                          data = bank, show.labels = TRUE))
+    expect_equal(result$outcome.label, "lbl")
+})
+
```

Comment:
This unit test still passes on the master branch which should reproduce the bug and the test should fail. After some testing it seems that the attributes are retained in this example. The attributes were lost in the Standard R test. The place where the attributes are lost is in `flipTransformations::Unclass`. In particular, https://github.com/Displayr/flipTransformations/blob/a726baf0eb31f83cbb3c1deb23d2e77500a96332/R/variable.R#L71-L81.

 If there are NAs then `unclass` is used which retains the attributes (this happens here with the bank data).

 However, if there are not any NAs (Standard R test example) then the attributes are lost since it enters the if block of the if else section linked above (i.e. coerced by using `as.character(as.numeric(x))`. I think perhaps the bug should be intercepted there and attributes of the label, name be retained?

Reply:
I have reverted my changes to flipRegression and made changes to flipTransformation and flipU to preserve attributes in AsNumeric. Please review those PRs first